modules/apps/object/object-test/src/testFunctional/macros/ValidateObjectsUpgrade.macro

```diff
@@ -0,0 +1,92 @@
+definition {
+
+	macro addEmailNotificationTemplate {
+		Navigator.openURL();
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
+
+		ObjectNotifications.addNewNotificationTemplate(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			notificationType = "${notificationType}",
+			subject = "${subject}");
+
+		if (isSet(objectLabelAttachment)) {
+			ObjectNotifications.addNotificationTemplateAttachment(objectLabel = "${objectLabelAttachment}");
+		}
+
+		PortletEntry.save();
+	}
+
+	macro editNotificationTemplate {
+		Navigator.openURL();
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
+
+		ObjectNotifications.editNotificationTemplate(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			externalReferenceCode = "${externalReferenceCode}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			notificationTemplateName = "${notificationTemplateName}",
+			subject = "${subject}");
+
+		PortletEntry.save();
+
+		ObjectNotifications.viewNotificationTemplateDetails(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			externalReferenceCode = "${externalReferenceCode}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			subject = "${subject}");
+	}
+
+	macro viewNotificationTemplate {
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
+
+		ObjectNotifications.viewNotificationTemplateDetails(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			subject = "${subject}");
+
+		AssertElementPresent(
+			key_dataSource = "Custom Object2",
+			locator1 = "ObjectNotifications#DATA_SOURCE_ATTACHMENTS");
+
+		AssertTextPresent(
+			locator1 = "ObjectNotifications#FIELD_ATTACHMENTS_OBJECT",
+			value1 = "Custom Field Attachment");
```

Comment:
Can we move this to the test level? If not, we shouldn't be hard coding here in case we want to use another custom object or field in the future.